### PR TITLE
DDF-3001 Update registry to insert subject information into requests.

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Updates the registry such that its requests include a subject. 
Also fixed issues in UpdateOperation, DeleteOperation and AbstractCswStore where the subject wasn't getting passed along with generated requests
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@emmberk @mcalcote 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).

[Core APIs](https://github.com/orgs/codice/teams/core-apis)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@stustison
#### How should this be tested? (List steps with links to updated documentation)
Test the following registry functions
-Publish
-Unpublish
-Publication update
-Subscription
#### What are the relevant tickets?
[DDF-3001](https://codice.atlassian.net/browse/DDF-3001)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java [hunk 1]
```diff
@@ -37,4 +43,30 @@ void buildPolicyMap(HashMap<String, Set<String>> policyMap,
...
+
+    /**
+     * Creates and returns a map of properties containing a security subject. If the correct subject
+     * type can not be obtained from the passed in subject or from the thread context no subject will
+     * be added to the properties.
+     * If the passed in subject is not compatible the method will fallback to trying to pull it from
+     * the thread context.
+     * @param subject The subject to be added. Can be {@code null}
+     */
+    Map<String, Serializable> getPropertiesWithSubject(Object subject) {
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java]: This method feels odd. It seems that while callers might want a `Map<String, Serializable>` that you're contorting yourself here to provide that. I'd think just returning the subject or null would be sufficient. I might be missing something this early in the review, however.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java [hunk 2]
```diff
@@ -37,4 +43,30 @@ void buildPolicyMap(HashMap<String, Set<String>> policyMap,
...
+        Map<String, Serializable> properties = new HashMap<>();
+        if(subject != null && subject instanceof Subject){
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java]: This test can be simplified to just the `instanceof` test. A `null` input for subject will return `false` for the `instanceof` test.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java [hunk 3]
```diff
@@ -37,4 +43,30 @@ void buildPolicyMap(HashMap<String, Set<String>> policyMap,
...
+            properties.put(SecurityConstants.SECURITY_SUBJECT, (Subject)subject);
+        } else {
+            try {
+                Object subjectFromContxt = SecurityUtils.getSubject();
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java]: Typo if you care.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java (hunk 1 above)
Comment (clockard) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java]: I agree. I wrote this method a few different ways but ended with this to minimize duplicating code. One of the ways was to just return the subject for a given operation. I will go back to that way of doing it.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java (hunk 2 above)
Comment (clockard) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java]: Good call

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java (hunk 3 above)
Comment (clockard) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java]: Oops. Fixed

**Comment by clockard** [comment]
build now

**Comment by stustison** [comment]
nit: You could probably have used `Collections.singletonMap(K, V)` for most of this to avoid the map creation.

**Comment by clockard** [comment]
Manually tested and verified.

**Pull request merged**